docs/resources/environment_drift_detection.md

```diff
@@ -0,0 +1,48 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "env0_environment_drift_detection Resource - terraform-provider-env0"
+subcategory: ""
+description: |-
+  
+---
+
+# env0_environment_drift_detection (Resource)
+
+
+
+## Example Usage
+
+```terraform
+data "env0_template" "example" {
+  name = "Template Name"
+}
+
+data "env0_project" "default_project" {
+  name = "Default Organization Project"
+}
+
+resource "env0_environment" "env" {
+  name        = "environment"
+  project_id  = data.env0_project.default_project.id
+  template_id = data.env0_template.example.id
+}
+
+resource "env0_environment_drift_detection" "drift" {
+  environment_id = env0_environment.env.id
+  cron = "0 4 * * *"
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- **cron** (String) Cron expression for scheduled drift detection of the environment
+- **environment_id** (String) The environment's id
+
+### Optional
+
+- **id** (String) The ID of this resource.
```

Comment:
why?

Reply:
If it really allows it you need o changed your logic using the id

Reply:
each tf resource have `id` which is the resource `id`
in this case the actual resource `id` doesn't matter but only the environment `id` which we set

Reply:
and it's optional since i didn't specify it as part of the schema